### ISSUE TITLE
Fatal exception when command is active and history is queried.

### ISSUE THREAD

**Issue by wphillipmoore** [wphillipmoore/efs-core#7]
This has been a very long standing bug, and it's something deep in the automatic object expansion code we're using to handle DateTime objects.  This one MUST be found and crushed.

```perl
efsops@nap8efst01$ efs history project efs core
DBIx::Class::InflateColumn::DateTime::catch {...} (): Error while inflating '0000-00-00 00:00:00' for endtime on EFS::DBI::Result::History=HASH(0x864d5c0): The 'month' parameter ("00") to DateTime::new did not pass the 'an integer between 1 and 12' callback: 
 at /efs/dist/perl5/DateTime/1.18/.exec/x86-64.rhel.6/5.20/lib/perl5/DateTime.pm line 197.
        DateTime::new(undef, "day", 00, "hour", 00, "second", 00, "year", 0000, ...) called at /efs/dist/perl5/DateTime-Format-Builder/0.81/common/lib/perl5/DateTime/Format/Builder/Parser/Regex.pm line 74
        DateTime::Format::Builder::Parser::Regex::make(DateTime::Format::Builder::Parser::Regex=HASH(0x87576f0), "0000-00-00 00:00:00", undef, HASH(0x8757438)) called at /efs/dist/perl5/DateTime-Format-Builder/0.81/common/lib/perl5/DateTime/Format/Builder/Parser/generic.pm line 82
        DateTime::Format::Builder::Parser::generic::__ANON__("DateTime::Format::MySQL", "0000-00-00 00:00:00", HASH(0x8831648)) called at /efs/dist/perl5/DateTime-Format-Builder/0.81/common/lib/perl5/DateTime/Format/Builder/Parser.pm line 279
        DateTime::Format::Builder::Parser::__ANON__("DateTime::Format::MySQL", "0000-00-00 00:00:00") called at /efs/dist/perl5/DateTime-Format-Builder/0.81/common/lib/perl5/DateTime/Format/Builder/Parser.pm line 63
        DateTime::Format::Builder::Parser::parse(DateTime::Format::Builder::Parser=HASH(0x8757a08), "DateTime::Format::MySQL", "0000-00-00 00:00:00") called at /efs/dist/perl5/DateTime-Format-Builder/0.81/common/lib/perl5/DateTime/Format/Builder.pm line 139
        DateTime::Format::Builder::__ANON__("DateTime::Format::MySQL", "0000-00-00 00:00:00") called at /efs/dist/perl5/DBIx-Class/0.082810/common/lib/perl5/DBIx/Class/InflateColumn/DateTime.pm line 201
        DBIx::Class::InflateColumn::DateTime::try {...} () called at /efs/dist/perl5/Try-Tiny/0.22/common/lib/perl5/Try/Tiny.pm line 79
        eval {...} called at /efs/dist/perl5/Try-Tiny/0.22/common/lib/perl5/Try/Tiny.pm line 72
        Try::Tiny::try(CODE(0x882ef80), Try::Tiny::Catch=REF(0x8830d90)) called at /efs/dist/perl5/DBIx-Class/0.082810/common/lib/perl5/DBIx/Class/InflateColumn/DateTime.pm line 207
        DBIx::Class::InflateColumn::DateTime::_flate_or_fallback(EFS::DBI::Result::History=HASH(0x864d5c0), "0000-00-00 00:00:00", HASH(0x633aeb8), "parse_%s") called at /efs/dist/perl5/DBIx-Class/0.082810/common/lib/perl5/DBIx/Class/InflateColumn/DateTime.pm line 212
        DBIx::Class::InflateColumn::DateTime::_inflate_to_datetime(EFS::DBI::Result::History=HASH(0x864d5c0), "0000-00-00 00:00:00", HASH(0x633aeb8)) called at /efs/dist/perl5/DBIx-Class/0.082810/common/lib/perl5/DBIx/Class/InflateColumn/DateTime.pm line 175
        DBIx::Class::InflateColumn::DateTime::__ANON__("0000-00-00 00:00:00", EFS::DBI::Result::History=HASH(0x864d5c0)) called at /efs/dist/perl5/DBIx-Class/0.082810/common/lib/perl5/DBIx/Class/InflateColumn.pm line 120
        DBIx::Class::InflateColumn::_inflated_column(EFS::DBI::Result::History=HASH(0x864d5c0), "endtime", "0000-00-00 00:00:00") called at /efs/dist/perl5/DBIx-Class/0.082810/common/lib/perl5/DBIx/Class/InflateColumn.pm line 171
        DBIx::Class::InflateColumn::get_inflated_column(EFS::DBI::Result::History=HASH(0x864d5c0), "endtime") called at (eval 526) line 2
        EFS::DBI::Result::History::endtime(EFS::DBI::Result::History=HASH(0x864d5c0)) called at /efs/dist/efs/core/3.004_005-build001/common/lib/perl5/EFS/CLI/History.pm line 231
        EFS::CLI::History::__ANON__("metaproj", "efs", "count", 100, "orderby", "id", "report", "default", "project", ...) called at /efs/dist/efs/core/3.004_005-build001/common/lib/perl5/EFS/CLI/History.pm line 383
        EFS::CLI::History::history("EFS::CLI::History", "metaproj", "efs", "project", "core", "report", "default", "orderby", "id", ...) called at /efs/dist/efs/core/3.004_005-build001/common/lib/perl5/EFS/CLI/Project.pm line 1173
        EFS::CLI::Project::history(undef, "name", "core", "metaproj", "efs") called at /efs/dist/efs/core/3.004_005-build001/common/lib/perl5/EFS/Request/CLI.pm line 145
        EFS::Request::CLI::try {...} () called at /efs/dist/perl5/Try-Tiny/0.22/common/lib/perl5/Try/Tiny.pm line 81
        eval {...} called at /efs/dist/perl5/Try-Tiny/0.22/common/lib/perl5/Try/Tiny.pm line 72
        Try::Tiny::try(CODE(0x860ac28), Try::Tiny::Catch=REF(0x860acb8)) called at /efs/dist/efs/core/3.004_005-build001/common/lib/perl5/EFS/Request/CLI.pm line 149
        EFS::Request::CLI::execute_no_authorization(EFS::Request::CLI=HASH(0x84b3c88)) called at /efs/dist/efs/core/3.004_005-build001/common/lib/perl5/EFS/Request/CLI.pm line 129
        EFS::Request::CLI::execute(EFS::Request::CLI=HASH(0x84b3c88)) called at /efs/dist/efs/core/3.004_005-build001/common/bin/.wrapped/efsd line 158
        main::HandleClient(IO::Socket::INET=GLOB(0x84c49d0)) called at /efs/dist/efs/core/3.004_005-build001/common/bin/.wrapped/efsd line 113 at /efs/dist/efs/core/3.004_005-build001/common/lib/perl5/EFS/CLI/History.pm line 231

```

**Comment by wphillipmoore** [comment]
First tidbit of research....     Check out ribasushi's first response in this bug report:

https://rt.cpan.org/Public/Bug/Display.html?id=99334

I am NOT loading InflateColumn::DateTime, and never have.  Just Core.  Let's try that, and see if the problem goes away.    Now, in order to test this, we need to be able to run a command that takes some time, in the background, and then run "efs history" to see if we can "see" the command.    In order to do THAT, I am going to add a new feature that I have been wanting for some time now.

(See next comment)

**Comment by wphillipmoore** [comment]
See https://github.com/wphillipmoore/efs-core/issues/8

We'll implement efs exec (trivial, really), and then use it to test the fix for THIS bug by running:

    efs exec efsserver .... 'sleep 10'

I'll most likely write the test suite to fork and run that command in the child, while the parent queries the history to invoke the bug.  Then, when the test suite exposes it, we hack until it goes away (which is how you fix all bugs, right?)

**Comment by wphillipmoore** [comment]
FWIW, adding InflateColumn::DateTime to load_components had no impact -- exact same error.

**Comment by wphillipmoore** [comment]
Fixed in commit fc177ab8abd2dcefac8c0601f204b83fdeeb745b

The key was specifying datetime_undef_if_invalid => 1in the column specification.    efs history any works fine, an shows you the active commands without raising an exception.

Damn...  Took me years to finally get around to crushing this one....
